diesel/src/query_builder/insert_statement.rs

```diff
@@ -176,7 +176,7 @@ where
 }
 
 #[cfg(feature = "sqlite")]
-impl<'a, T, U, Op> ExecuteDsl<SqliteConnection> for InsertStatement<T, &'a [U], Op>
```

Comment:
This is a breaking change, because this will break code that uses this as a trait bound

Reply:
This is not a breaking change. There is no way to construct this type in public API other than by constructing it by passing a slice to `values`. All of that continues to work.

Reply:
I guess I can leave this impl in place -- it's straight up impossible to construct this type though

Reply:
It is not impossible, because I've a codebase where something like this exists:
```rust
trait MyOwnInsert<&'insert, T, Res, E> {
    type Ret;
    fn insert(&'insert self, conn: &SqliteConnection) -> QueryResult<Self::Ret>;
}
```
Now if I implement this trait for `C` where `C: AsRef<[E]> I need (amongst other) the following trait bound:
```rust
 InsertStatement<T, <&'insert [E] as Insertable<T>>::Values>: ExecuteDsl<SqliteConnection>,
```
(Note that this only uses code from our public API)

Reply:
The code you just pasted would continue to work, and is the intended way to do that.